specs/core/0_beacon-chain.md

```diff
@@ -1214,30 +1215,44 @@ def verify_bitfield(bitfield: bytes, committee_size: int) -> bool:
     return True
 ```
 
-### `verify_slashable_attestation`
+### `convert_to_standalone`
 
 ```python
-def verify_slashable_attestation(state: BeaconState, slashable_attestation: SlashableAttestation) -> bool:
+def convert_to_standalone(state: BeaconState, attestation: Attestation):
     """
-    Verify validity of ``slashable_attestation`` fields.
+    Convert an attestation to (almost) standalone-verifiable form
     """
-    if slashable_attestation.custody_bitfield != b'\x00' * len(slashable_attestation.custody_bitfield):  # [TO BE REMOVED IN PHASE 1]
-        return False
+    return StandaloneAttestation(
+        validator_indices=get_attestation_participants(state, attestation.data, attestation.aggregation_bitfield),
+        data=attestation.data,
+        custody_bitfield=attestation.custody_bitfield,
```

Comment:
To confirm with @vbuterin:

1. I *assumed* that before this conversion, the committee size of this `Attestation.custody_bitfield` is the committee of the given crosslink committee - but actually, the size of `Attestation.custody_bitfield` is not restricted in the `dev` branch.
2. And after this convertion, in `verify_standalone_attestation`, it expects that the size of `StandaloneAttestation.custody_bitfield` is `len(StandaloneAttestation.validator_indices)`?

The benefit of having `len(Attestation.custody_bitfield) == len(Attestation.aggregation_bitfield) == len(crosslink_committee)` is that we can simply merge the bitfields with `or` operator when aggregating 
 many `Attestation`s. If the size of `len(Attestation.custody_bitfield) != len(crosslink_committee)`, it seems it would add additional overhead to generate the new `custody_bitfield`?

Reply:
> I assumed that before this conversion, the committee size of this Attestation.custody_bitfield is the committee of the given crosslink committee - but actually, the size of Attestation.custody_bitfield is not restricted in the dev branch.

An attestation committee will not exceed 4096 already because that's the "all ETH is validating" threshold. In any case, having the same threshold in both regular attestations and standalone attestations seems necessary for slashing to be possible in any case.

> it expects that the committee
size of StandaloneAttestation.custody_bitfield is len(StandaloneAttestation.validator_indices)?

You are right that the proposed change does increase strictness because the current approach requires the custody bitfield to be _any_ quantity of zero bytes. I would argue again that the current approach is wrong and the new approach is correct.